src/protocolsupport/zplatform/impl/glowstone/GlowStoneMiscUtils.java

```diff
@@ -200,6 +207,12 @@ public void enableCompression(ChannelPipeline pipeline, int compressionThreshold
 		pipeline.addAfter(GlowStoneChannelHandlers.FRAMING, "compression", new CompressionHandler(compressionThreshold));
 	}
 
+	@Override
+	public void enableEncryption(ChannelPipeline pipeline, SecretKey key, boolean fullEncryption) {
+		pipeline.addBefore(GlowStoneChannelHandlers.FRAMING, ChannelHandlers.ENCRYPT, new PacketEncrypter(MinecraftEncryption.getCipher(Cipher.ENCRYPT_MODE, key)));
+		pipeline.addBefore(ChannelHandlers.ENCRYPT, ChannelHandlers.DECRYPT, new PacketDecrypter(MinecraftEncryption.getCipher(Cipher.DECRYPT_MODE, key)));
```

Comment:
You forgot to check if encrpytion is full

Reply:
Is full encryption a cross-platform thing? If so, what's the pipeline handler ordering for it?

Reply:
Full encryption means that both encryption handlers are present. Not full means that only decryption handler should be present.